src/hotspot/share/opto/ifnode.cpp

```diff
@@ -114,16 +114,33 @@ static Node* split_if(IfNode *iff, PhaseIterGVN *igvn) {
   if( !t->singleton() ) return NULL;
 
   // No intervening control, like a simple Call
-  Node *r = iff->in(0);
-  if( !r->is_Region() ) return NULL;
-  if (r->is_Loop()) return NULL;
-  if( phi->region() != r ) return NULL;
+  Node* r = iff->in(0);
+  if (!r->is_Region() || r->is_Loop() || phi->region() != r) {
```

Comment:
May be also add ` || r->as_Region()->is_copy() || igvn->_worklist.member(r)` checks.

Reply:
Could we then miss a split if opportunity when bailing out with `igvn->_worklist.member(r)`? Should we readd the `If` node to the worklist in this case? We could also think about adding `igvn->_worklist.member(r)` further down when we are really sure that we could do a split if.

Reply:
My thinking is that Region could go away after it is processed (when input[s] are dead/top). Why bother to split through it?

I am concern that you are checking dead inputs by hand in these changes - input's transformation to Top should put Region node on worklist. Can you do experiment without your dead inputs check and only checking worklist?

I think the issue (as usual) is the order of processing of worklist. That is why I suggested this check.
Yes, you can delay worklist checking to see if we should put If node back on worklist only when we can do split if.

Reply:
You're right, this additional worklist check should then be enough and is more clean. I reverted the fix and added a worklist check at the latest possible location. This works for repeated runs with `StressIGVN` + `RepeatCompilation`. I also included the `is_copy()` check.